File: oop/code/oop/src/main/java/com/scaler/lld/bird/interfaces/Penguin.java
package com.scaler.lld.bird.interfaces;

public class Penguin extends Bird implements Swimmable {
    public Penguin(Double weight, String colour, String size, String beakType, BirdType type) {
        super(weight, colour, size, beakType, type);
    }

    @Override
    public void swim() {
        System.out.println("Penguin is swimming.");
    }
}

File: oop/code/python/oop/SOLID/bird/Penguin.py
from Bird import *
from BirdType import *
from interfaces.SwimmableInterface import *
from interfaces.FlyingBehaviourInterface import *

class Penguin(Bird, SwimmableInterface):
    def __init__(self, weight: int, colour: str, size: int, beakType: str, birdType: BirdType) -> None:
        super().__init__(weight, colour, size, beakType, birdType)
    
    def fly(self) -> None:
        self.__flyingBehaviour.makeFly()
        
    def makeSound(self) -> None:
        ...